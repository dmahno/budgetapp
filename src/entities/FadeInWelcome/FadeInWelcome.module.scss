.fadeInWelcome {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #fff;
  background: linear-gradient(
      135deg,
      rgb(249, 214, 88) 0%,
      rgb(73, 110, 11) 55.725%,
      rgb(102, 79, 27) 91.603%
    ),
    rgb(255, 255, 255);
  font-size: 2rem;
  opacity: 1;
  animation: fadeInOut 3s forwards;
  overflow: hidden;
  padding: $indent-m;

  &::before {
    content: '';
    position: absolute;
    top: -10px;
    left: -10px;
    width: calc(100% + 20px);
    height: calc(100% + 20px);
    background: url('../../shared/assets/images/loginBg.png') no-repeat center
      center;
    background-size: cover;
    filter: blur(8px);
    z-index: -1;
  }
}

@keyframes fadeInOut {
  0%,
  20%,
  85% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

.title {
  font-weight: 600;
  font-size: 44px;
  @include mobile {
    font-size: 32px;
  }
}
