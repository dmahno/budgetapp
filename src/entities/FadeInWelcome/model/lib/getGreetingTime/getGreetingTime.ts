export const getGreetingTime = () => {
  const hours = new Date().getHours();

  if (hours >= 0 && hours < 5) return 'Доброй ночи';
  if (hours >= 5 && hours < 12) return 'Доброе утро';
  if (hours >= 12 && hours < 18) return 'Добрый день';
  return 'Добрый вечер';
};
