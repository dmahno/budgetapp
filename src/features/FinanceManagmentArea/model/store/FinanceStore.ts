// FinanceStore.ts
import {makeAutoObservable} from 'mobx';
import {message} from 'antd';

import {getDatabase} from '../lib/getDatabase/getDatabase';

interface IOperation {
  id?: number;
  amount: number;
  description: string;
  type: 'add' | 'exclude';
  date: Date;
}

export class FinanceStore {
  operationsList: IOperation[] = [];

  constructor() {
    makeAutoObservable(this);
    this.loadOperations();
  }

  async loadOperations() {
    const database = await getDatabase();
    this.operationsList = await database.getAll('operations');
  }

  async addOperation(
    amount: number,
    description: string,
    type: 'add' | 'exclude',
  ) {
    if (type === 'exclude' && amount > this.totalSum) {
      message.error(
        `Недостаточно средств. Максимальная сумма для списания: ${this.totalSum} ₽`,
      );
      return;
    }

    const database = await getDatabase();
    const newOperation: IOperation = {
      amount,
      description,
      type,
      date: new Date(),
    };

    const operationId = (await database.add(
      'operations',
      newOperation,
    )) as number;
    this.operationsList.push({...newOperation, id: operationId});
  }

  async deleteOperation(operationId: number) {
    const database = await getDatabase();
    await database.delete('operations', operationId);
    this.operationsList = this.operationsList.filter(
      (operation) => operation.id !== operationId,
    );
  }

  get totalSum() {
    return this.operationsList.reduce((accumulatedSum, operation) => {
      return operation.type === 'add'
        ? accumulatedSum + operation.amount
        : accumulatedSum - operation.amount;
    }, 0);
  }
}

export const financeStore = new FinanceStore();
