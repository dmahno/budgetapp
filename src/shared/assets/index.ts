export type TImageKeys = '1.jpg' | '2.jpg' | '3.jpg' | '4.jpg' | '5.jpg';

import image1 from './images/1.jpg';
import image2 from './images/2.jpg';
import image3 from './images/3.jpg';
import image4 from './images/4.jpg';
import image5 from './images/5.jpg';

export const images: Record<TImageKeys, string> = {
  '1.jpg': image1,
  '2.jpg': image2,
  '3.jpg': image3,
  '4.jpg': image4,
  '5.jpg': image5,
};
