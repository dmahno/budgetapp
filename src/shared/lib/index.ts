export {formatNumber} from './formatNumber/formatNumber';
export {parseNumber} from './parseNumber/parseNumber';
