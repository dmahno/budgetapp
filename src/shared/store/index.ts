export * from './AuthStore/AuthStore';
export * from './ThemeStore/ThemeStore';
