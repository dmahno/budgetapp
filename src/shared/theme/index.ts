export {configDarkTheme} from './tokens/configDarkTheme';
export {configLightTheme} from './tokens/configLightTheme';
