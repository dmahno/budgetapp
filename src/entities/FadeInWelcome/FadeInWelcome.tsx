import {observer} from 'mobx-react-lite';

import {authStore} from 'shared/store';

import {getGreetingTime} from './model/lib/getGreetingTime/getGreetingTime';
import styles from './FadeInWelcome.module.scss';

export const FadeInWelcome = observer(() => {
  const {userName, isWelcome} = authStore;

  if (!isWelcome) return null;
  const firstName = userName ? userName.split(' ')[0] : '';

  return (
    <div className={styles.fadeInWelcome}>
      <h1 className={styles.title}>
        {getGreetingTime()}
        {firstName && `, ${firstName}`}! <br /> Входим в приложение.
      </h1>
    </div>
  );
});
