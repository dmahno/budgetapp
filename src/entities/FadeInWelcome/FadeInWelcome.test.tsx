import {render, screen} from '@testing-library/react';
import {describe, it, expect, vi, beforeEach} from 'vitest';

import {authStore} from 'shared/store';

import {FadeInWelcome} from './FadeInWelcome';
import * as greetingTimeModule from './model/lib/getGreetingTime/getGreetingTime';

vi.mock('shared/store', () => ({
  authStore: {
    userName: '',
    isWelcome: false,
  },
}));

describe('FadeInWelcome', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders null when isWelcome is false', () => {
    authStore.isWelcome = false;
    render(<FadeInWelcome />);
    expect(screen.queryByText(/Входим в приложение/)).toBeNull();
  });

  it('renders greeting with user first name when isWelcome is true and userName exists', () => {
    authStore.isWelcome = true;
    authStore.userName = 'Дмитрий Морозов';
    vi.spyOn(greetingTimeModule, 'getGreetingTime').mockReturnValue(
      'Доброе утро',
    );
    render(<FadeInWelcome />);

    const greetingText = screen.getByText(
      /Доброе утро, Дмитрий! Входим в приложение./,
    );
    expect(greetingText).toBeInTheDocument();
  });

  it('renders greeting without user name when isWelcome is true and userName is empty', () => {
    authStore.isWelcome = true;
    authStore.userName = '';
    vi.spyOn(greetingTimeModule, 'getGreetingTime').mockReturnValue(
      'Добрый день',
    );
    render(<FadeInWelcome />);

    const greetingText = screen.getByText(/Добрый день! Входим в приложение./);
    expect(greetingText).toBeInTheDocument();
  });

  it('renders correctly based on different greeting times', () => {
    authStore.isWelcome = true;
    authStore.userName = 'Дмитрий Морозов';

    vi.spyOn(greetingTimeModule, 'getGreetingTime').mockReturnValueOnce(
      'Доброе утро',
    );
    render(<FadeInWelcome />);
    expect(
      screen.getByText(/Доброе утро, Дмитрий! Входим в приложение./),
    ).toBeInTheDocument();

    vi.spyOn(greetingTimeModule, 'getGreetingTime').mockReturnValueOnce(
      'Добрый вечер',
    );
    render(<FadeInWelcome />);
    expect(
      screen.getByText(/Добрый вечер, Дмитрий! Входим в приложение./),
    ).toBeInTheDocument();
  });
  it('matches the snapshot', () => {
    const {asFragment} = render(<FadeInWelcome />);
    expect(asFragment()).toMatchSnapshot();
  });
});
