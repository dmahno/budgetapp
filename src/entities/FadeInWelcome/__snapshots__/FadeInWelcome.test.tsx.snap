// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FadeInWelcome > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="_fadeInWelcome_270f4f"
  >
    <h1
      class="_title_270f4f"
    >
      Добрый день, Дмитрий! 
      <br />
       Входим в приложение.
    </h1>
  </div>
</DocumentFragment>
`;
